oauth2client/_helpers.py

```diff
@@ -300,3 +295,55 @@ def _urlsafe_b64decode(b64string):
     b64string = _to_bytes(b64string)
     padded = b64string + b'=' * (4 - len(b64string) % 4)
     return base64.urlsafe_b64decode(padded)
+
+
+def update_query_params(uri, params):
+    """Updates a URI with new query parameters.
+
+    If a given key from ``params`` is repeated in the ``uri``, then
```

Comment:
Sorcerery!

These are really strange semantics; is there some resource to which you can link that explains why this is correct or useful? (Yes, there's a lot I don't know about the web.)